src/App.js

```diff
@@ -26,25 +29,38 @@ class PanelButtons extends Component {
     setCurrentPanel: PropTypes.func,
     onContinue: PropTypes.func,
     startSaveTrainedModel: PropTypes.func,
-    dataToSave: PropTypes.object
+    dataToSave: PropTypes.object,
+    saveStatus: PropTypes.string
   };
 
   onClickPrev = () => {
     this.props.setCurrentPanel(this.props.panelButtons.prev.panel);
   };
 
   onClickNext = () => {
-    if (this.props.panelButtons.next.panel === "save") {
-      this.props.startSaveTrainedModel(this.props.dataToSave);
-    } else if (this.props.panelButtons.next.panel === "continue") {
+    if (["continue", "finish"].includes(this.props.panelButtons.next.panel)) {
       this.props.onContinue();
-    } else {
-      this.props.setCurrentPanel(this.props.panelButtons.next.panel);
+      return;
+    }
+
+    if (this.props.currentPanel === "saveModel") {
+      this.props.startSaveTrainedModel(this.props.dataToSave);
+      return;
     }
+
+    this.props.setCurrentPanel(this.props.panelButtons.next.panel);
   };
 
   render() {
-    const { panelButtons } = this.props;
+    const { panelButtons, saveStatus } = this.props;
+
+    let loadStatus = !["success", "failure"].includes(saveStatus)
```

Comment:
Does this imply `saveStatus` is `"started"` or is there another possibility?

Reply:
`saveStatus` by default is `undefined`. This covers the cases where the save hasn't finished and it'd still work if we wanted to add more in between states in the future. :crossed_fingers: What do you think?

Reply:
I think we could refactor this section of code a bit to move some of the specific knowledge about possible values of `saveStatus`, and what to do with it, out of the UI and into a Redux helper function or two.  Then we simply have a prop (or two) which tells us whether we show a spinner, whether we show a message (and what it is), and then we avoid rendering `{undefined}` for nothing as an added bonus.  This also means that if we add another state to `saveStatus` later, we don't need to hardcore references to it in our views, and rather centralise that management in `redux.js`.

Reply:
Good call! I refactored it. Let me know if this update is on track with what you were thinking.